ce/ce/session.ts

```diff
@@ -125,139 +118,31 @@ export class Session {
-    const r = this.registries.getRegistry(registryLocation.toString());
-    if (r) {
-      return r;
-    }
-
-    // not already loaded
-    registryLocation = this.parseUri(registryLocation);
-
-    switch (registryKind) {
-
-      case 'artifact':
-        switch (registryLocation.scheme) {
-          case 'https':
-            return this.registries.add(new RemoteRegistry(this, registryLocation));
-
-          case 'file':
-            return this.registries.add(new LocalRegistry(this, registryLocation));
-
-          default:
-            throw new Error(i`Unsupported registry scheme '${registryLocation.scheme}'`);
-        }
+  parseLocation(location: string): Uri {
+    // Drive letter, absolute Unix path, or drive-relative windows path, treat as a file
+    if (/^[A-Za-z]:/.exec(location) || location.startsWith('/') || location.startsWith('\\')) {
+      return this.fileSystem.file(location);
     }
-    throw new Error(i`Unsupported registry kind '${registryKind}'`);
-  }
-
-  async isLocalRegistry(location: Uri | string): Promise<boolean> {
-    location = this.parseUri(location);
-
-    if (location.scheme === 'file') {
-      if (await isIndexFile(location)) {
-        return true;
-      }
 
-      if (await location.isDirectory()) {
-        const index = location.join(registryIndexFile);
-        if (await isIndexFile(index)) {
-          return true;
-        }
-        const s = this;
-        let result = false;
-        const q = new Queue();
-
-        // still could be a folder of artifact files
-        // eslint-disable-next-line no-inner-declarations
-        async function process(folder: Uri) {
-          for (const [entry, type] of await folder.readDirectory()) {
-            if (result) {
-              return;
-            }
-
-            if (type & FileType.Directory) {
-              await process(entry);
-              continue;
-            }
-
-            if (type & FileType.File && entry.path.endsWith('.json')) {
-              void q.enqueue(async () => { result = result || await isMetadataFile(entry, s); });
-            }
-          }
-        }
-        await process(location);
-        await q.done;
-        return result; // whatever we guess, we'll use
-      }
-      return false;
+    // Has a scheme, it's a URI
+    if (/^(\w+):/.exec(location)) {
+      return this.fileSystem.parse(location);
     }
 
-    return false;
-  }
-
-  async isRemoteRegistry(location: Uri | string): Promise<boolean> {
-    return this.parseUri(location).scheme === 'https';
+    // Relative path
+    return this.currentDirectory.join(location);
   }
 
-  parseName(id: string): [string, string] {
-    const parts = id.split(':');
-    switch (parts.length) {
-      case 0:
-        throw new Error(i`Invalid artifact id '${id}'`);
-      case 1:
-        return ['default', id];
+  async loadDefaultRegistryResolver(manifest: ProjectManifest | undefined) : Promise<RegistryResolver> {
```

Comment:
The problem is that makes `undefined` checks in callers painful since ProjectManifest is one of the things likely to be `undefined`. For example:

```diff
diff --git a/ce/ce/cli/commands/acquire.ts b/ce/ce/cli/commands/acquire.ts
index 23253dfb..95a227ed 100644
--- a/ce/ce/cli/commands/acquire.ts
+++ b/ce/ce/cli/commands/acquire.ts
@@ -44,7 +44,8 @@ export class AcquireCommand extends Command {
       return false;
     }

-    const resolver = await session.loadDefaultRegistryResolver(await this.project.manifest);
+    const project = await this.project.manifest;
+    const resolver = (await project?.buildRegistryResolver())?.combineWith(session.globalRegistryResolver) ?? session.globalRegistryResolver;
     const resolved = await selectArtifacts(session, new Map(this.inputs.map((v, i) => [v, versions[i] || '*'])), resolver, 1);        
     if (!resolved) {
       debug('No artifacts selected - stopping');
```

Reply:
I suppose I could flip combineWith's sense since session.globalRegistryResolver won't be undefined...